### PR TITLE
Adding entries to Pod /etc/hosts with HostAliases for ID localization.

### PR DESCRIPTION
**Pull request by irvifa**
This address #15341

### CODE REVIEW

**Comment by sftim** [comment]
I can't properly review Bahasa Indonesia, sorry!

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+Secara default, file `hosts` hanya berisikan _boilerplate_ alamat IP IPv4 and IPv6 seperti 
+`localhost` dan hostname dari Pod itu sendiri.
+
+## Menambahkan _Entry_ Tambahan dengan HostAliases
+
+Selain _boilerplate_ default, kita dapat menambahkan _entry_ pada file 
+`hosts` untuk melakukan resolusi `foo.local`, `bar.local` pada `127.0.0.1` dan `foo.remote`,
+`bar.remote` pada `10.1.2.3`, kita dapat melakukannya dengan cara menambahkan 
+HostAliases pada Pod dibawah _field_ `.spec.hostAliases`:
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
HostAliases pada Pod di bawah _field_ `.spec.hostAliases`:
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+
+{{< codenew file="service/networking/hostaliases-pod.yaml" >}}
+
+Pod ini kemudian dapat di-_start_ dengan perintah berikut:
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
Pod ini kemudian dapat dihidupkan dengan perintah berikut:
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+```shell
+kubectl get pod --output=wide
+```
+
+```shell
+NAME                           READY     STATUS      RESTARTS   AGE       IP              NODE
+hostaliases-pod                0/1       Completed   0          6s        10.200.0.5      worker0
+```
+
+File hosts yang ada akan tampak sebagai berikut:
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
Berkas `hosts` yang ada akan tampak sebagai berikut:
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+fe00::1	ip6-allnodes
+fe00::2	ip6-allrouters
+10.200.0.5	hostaliases-pod
+
+# Entries added by HostAliases.
+127.0.0.1	foo.local	bar.local
+10.1.2.3	foo.remote	bar.remote
+```
+
+Dengan tambahan _entry_ yang telah dispesifikasikan sebelumnya.
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
Dengan tambahan entri yang telah dispesifikasikan sebelumnya.
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+
+## Kenapa Kubelet Melakukan Mekanisme Manajemen File Hosts?
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
## Kenapa Kubelet Melakukan Mekanisme Manajemen Berkas `Hosts`?
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
+Kubelet [melakukan proses manajemen](https://github.com/kubernetes/kubernetes/issues/14633) 
+file `hosts` untuk setiap container yang ada pada Pod untuk mencegah Docker melakukan
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
berkas `hosts` untuk setiap kontainer yang ada pada Pod untuk mencegah Docker melakukan 
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 7]
```diff
@@ -0,0 +1,131 @@
...
+10.1.2.3	foo.remote	bar.remote
+```
+
+Dengan tambahan _entry_ yang telah dispesifikasikan sebelumnya.
+
+## Kenapa Kubelet Melakukan Mekanisme Manajemen File Hosts?
+
+Kubelet [melakukan proses manajemen](https://github.com/kubernetes/kubernetes/issues/14633) 
+file `hosts` untuk setiap container yang ada pada Pod untuk mencegah Docker melakukan 
+[modifikasi](https://github.com/moby/moby/issues/17190) pada file tersebut
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
[modifikasi](https://github.com/moby/moby/issues/17190) pada berkas tersebut 
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 8]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+Dengan tambahan _entry_ yang telah dispesifikasikan sebelumnya.
+
+## Kenapa Kubelet Melakukan Mekanisme Manajemen File Hosts?
+
+Kubelet [melakukan proses manajemen](https://github.com/kubernetes/kubernetes/issues/14633) 
+file `hosts` untuk setiap container yang ada pada Pod untuk mencegah Docker melakukan 
+[modifikasi](https://github.com/moby/moby/issues/17190) pada file tersebut 
+setelah kontainer di-_start_.
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
setelah kontainer dihidupkan.
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+
+Karena sifat dari file tersebut yang secara otomatis di-_manage_,
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
Karena sifat dari berkas tersebut yang secara otomatis dikelola, 
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 10]
```diff
@@ -0,0 +1,131 @@
...
+
+## Kenapa Kubelet Melakukan Mekanisme Manajemen File Hosts?
+
+Kubelet [melakukan proses manajemen](https://github.com/kubernetes/kubernetes/issues/14633) 
+file `hosts` untuk setiap container yang ada pada Pod untuk mencegah Docker melakukan 
+[modifikasi](https://github.com/moby/moby/issues/17190) pada file tersebut 
+setelah kontainer di-_start_.
+
+Karena sifat dari file tersebut yang secara otomatis di-_manage_, 
+semua hal yang didefinisikan oleh pengguna akan di-_overwite_ ketika file
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
semua hal yang didefinisikan oleh pengguna akan ditimpa (_overwite_) ketika berkas
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 11]
```diff
@@ -0,0 +1,131 @@
...
+## Kenapa Kubelet Melakukan Mekanisme Manajemen File Hosts?
+
+Kubelet [melakukan proses manajemen](https://github.com/kubernetes/kubernetes/issues/14633) 
+file `hosts` untuk setiap container yang ada pada Pod untuk mencegah Docker melakukan 
+[modifikasi](https://github.com/moby/moby/issues/17190) pada file tersebut 
+setelah kontainer di-_start_.
+
+Karena sifat dari file tersebut yang secara otomatis di-_manage_, 
+semua hal yang didefinisikan oleh pengguna akan di-_overwite_ ketika file 
+`hosts` di-_mount_ kembali oleh Kubelet ketika sebuah kontainer di-_restart_
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
`hosts` di-_mount_ kembali oleh Kubelet ketika ada kontainer yang di-_restart_ 
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 12]
```diff
@@ -0,0 +1,131 @@
...
+Kubelet [melakukan proses manajemen](https://github.com/kubernetes/kubernetes/issues/14633) 
+file `hosts` untuk setiap container yang ada pada Pod untuk mencegah Docker melakukan 
+[modifikasi](https://github.com/moby/moby/issues/17190) pada file tersebut 
+setelah kontainer di-_start_.
+
+Karena sifat dari file tersebut yang secara otomatis di-_manage_, 
+semua hal yang didefinisikan oleh pengguna akan di-_overwite_ ketika file 
+`hosts` di-_mount_ kembali oleh Kubelet ketika sebuah kontainer di-_restart_ 
+atau Pod di-_schedule_ ulang. Dengan demikian tidak dianjurkan untuk 
+memodifikasi file tersebut secara langsung.
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]:
```suggestion
memodifikasi berkas tersebut secara langsung.
```

File: content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md [hunk 13]
```diff
@@ -0,0 +1,131 @@
...
+```
+
+File hosts yang ada akan tampak sebagai berikut:
+
+```shell
+kubectl exec nginx -- cat /etc/hosts
+```
+
+```none
+# Kubernetes-managed hosts file.
```
Comment (girikuncoro) [inline content/id/docs/concepts/services-networking/add-entries-to-pod-etc-hosts-with-host-aliases.md]: English comment should be translated, suggestion:

Berkas hosts yang dikelola Kubernetes

**Comment by irvifa** [comment]
@girikuncoro Addressed in the latest commit.

**Comment by girikuncoro** [comment]
sorry for the delay in getting this in, great work! 🎉 

/lgtm
/approve

**Pull request merged**